client/state/selectors/get-sites.js

```diff
@@ -20,7 +20,7 @@ export default createSelector(
 	( state ) => {
 		const primarySiteId = getPrimarySiteId( state );
 		const [ primarySite, sites ] = partition( state.sites.items, { ID: primarySiteId } );
-		return map( concat( primarySite, sites ), site => getSite( state, site.ID ) );
+		return map( concat( primarySite, sortBy( sites, [ 'name', 'URL' ] ) ), site => getSite( state, site.ID ) );
```

Comment:
Before a site with no name would be first in the list. Not saying that's what we should do.